File: android/app/src/main/kotlin/ai/gowtam/oak/ui/orbs/OrbEngine.kt
package ai.gowtam.oak.ui.orbs

object OrbEngine {
    internal fun resolve(state: OrbState, size: OrbSize): OrbResolved = OrbPresets.resolve(state, size)

    fun frame(state: OrbState, size: OrbSize, t: Double): OrbFrame {
        val resolved = OrbPresets.resolve(state, size)
        return OrbModes.frame(resolved.mode, size.px.toDouble(), t, resolved.opts)
    }
}

File: ios/OakApp/UI/Orbs/OrbEngine.swift
import Foundation

enum OrbEngine {
  static func resolve(_ state: OrbState, size: OrbSize) -> OrbPresets.Resolved {
    OrbPresets.resolve(state, size: size)
  }

  /// Geometry at instant `t` (already speed-scaled by the view, or raw for golden tests).
  static func frame(state: OrbState, size: OrbSize, t: Double) -> OrbFrame {
    let resolved = OrbPresets.resolve(state, size: size)
    return OrbModes.frame(mode: resolved.mode, size: Double(size.rawValue), t: t, opts: resolved.opts)
  }
}
